patches/0001-ffmpeg-Use-royalty-free-libfdk-aac-for-AAC-playback.patch

```diff
@@ -587,23 +1700,43 @@ index 1908ba19e..920b22bfa 100644
      &ff_flac_demuxer,
      &ff_matroska_demuxer,
      &ff_mov_demuxer,
+diff --git a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
+index 0f5c7b1d74..bc43b980d5 100644
+--- a/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
++++ b/third_party/ffmpeg/chromium/config/Chromium/linux/x64/libavutil/ffversion.h
+@@ -1,5 +1,5 @@
+ /* Automatically generated by version.sh, do not manually edit! */
+ #ifndef AVUTIL_FFVERSION_H
+ #define AVUTIL_FFVERSION_H
+-#define FFMPEG_VERSION "N-100481-g0493b979df"
++#define FFMPEG_VERSION "4.3.git"
+ #endif /* AVUTIL_FFVERSION_H */
 diff --git a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
-index 75333a2c9..7218d99ed 100755
+index 75333a2c9f..87de071cff 100755
 --- a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
 +++ b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
-@@ -910,11 +910,20 @@ def ConfigureAndBuild(target_arch, target_os, host_os, host_arch, parallel_jobs,
+@@ -910,11 +910,29 @@ def ConfigureAndBuild(target_arch, target_os, host_os, host_arch, parallel_jobs,
            '--ar=cygwin-wrapper lib',
        ])
  
-+  # Depend on royalty-free libfdk-aac for AAC support.
 +  if target_os == 'linux':
-+    configure_flags['Common'].extend([
-+        # Build against libfdk-aac and enable it always.
++    # Depend on openh264 for H.264 support.
++    configure_flags['Chromium'].extend([
++        '--enable-libopenh264',
++        '--enable-decoder=libopenh264',
++        '--disable-decoder=h264',
++    ])
++
++    # Depend on royalty-free libfdk-aac for AAC support.
++    configure_flags['Chrome'].extend([
```

Comment:
I don't think we want that, we want fdk-aac enabled in the free (`Chromium`) configuration at least - actually thinking about it I think we want fdk-aac in the Chromium config and use the built-in AAC decoder (ff_aac_decoder) in the Chrome (non-free) config? As is now (current master) both configurations are using fdk-aac.

Reply:
fdk-aac only supports a subset of AAC profiles so I think we should do the same as we do for openh264 and only enable it in the free configuration.

Reply:
(fwiw, the behaviour suggested above is actually what our current tests expect on Endless OS).

Reply:
@andrunko I don't think that's correct, the fdk-aac license says

> NO EXPRESS OR IMPLIED LICENSES TO ANY PATENT CLAIMS, including without limitation the patents of Fraunhofer, ARE GRANTED BY THIS SOFTWARE LICENSE. Fraunhofer provides no warranty of patent non-infringement with respect to this software.
>
> You may use this FDK AAC Codec software or modifications thereto only for purposes that are authorized by appropriate patent licenses.

which probably shouldn't be in the free builds?

Reply:
It's also mentioned [on the FFmpeg page](https://trac.ffmpeg.org/wiki/Encode/AAC):

> The license of libfdk_aac is not compatible with GPL, so the GPL does not permit distribution of binaries containing incompatible code when GPL-licensed code is also included. Therefore this encoder have been designated as "non-free", and you cannot download a pre-built ffmpeg that supports it.

Reply:
and in their news page

> Licensing has always been an issue with encoding AAC audio as most of the encoders have had a license making FFmpeg unredistributable if compiled with support for them. The fact that there now exists a fully open and truly free AAC encoder integrated directly within the project means a lot to those who wish to use accepted and widespread standards.

Maybe that was just backwards?

Reply:
Aiui we want to enable fdk-aac in the free configuration but I don't see a point in enabling it in the non-free configuration. Fwiw, libfdk-aac is shipped in the `org.freedesktop.Platform` runtime. 

In any case we can do this change (to only enable fdk-aac in the free config) in a separate PR if you prefer, but we should at least enable it in the free builds (current status quo) for now to avoid regressions.

If we make this change to only enable fdk-aac in the free config we will also need to adjust `CheckNonFreeMimeTypesOnFlatpak()` on `media/base/mime_util_internal.cc` to check for `ff_aac_decoder` when deciding if AAC is supported (+ adjust the commit msg for that patch).

Reply:
@refi64 Just to follow-up, looks like we still want to keep fdk-aac for non-free builds (as well as free builds) as we still plan to ship the `.Codecs` extension on some EOS images without incurring AAC royalties. @ramcq @wjt is that correct?

Reply:
Yes, please can we keep only patent-unencumbered fdk-aac in `.Codecs` too, and not a patent-encumbered AAC implementation?